src/Identity/IdentityServer/CustomTokenRequestValidator.cs

```diff
@@ -202,25 +199,28 @@ public async Task ValidateAsync(CustomTokenRequestValidationContext context)
     /// <summary>
     /// Used to create a list of all possible ways the newly authenticated user can decrypt their vault contents
     /// </summary>
-    private async Task<UserDecryptionOptions> CreateUserDecryptionOptionsAsync(SsoConfigurationData ssoConfigurationData, User user)
+    private async Task<UserDecryptionOptions> CreateUserDecryptionOptionsAsync(SsoConfigurationData? ssoConfigurationData, User user)
     {
-        var userDecryptionOption = new UserDecryptionOptions();
+        var userDecryptionOption = new UserDecryptionOptions
+        {
+            HasMasterPassword = !string.IsNullOrEmpty(user.MasterPassword)
+        };
+
         if (ssoConfigurationData is { MemberDecryptionType: MemberDecryptionType.KeyConnector } && !string.IsNullOrEmpty(ssoConfigurationData.KeyConnectorUrl))
         {
             // KeyConnector makes it mutually exclusive
```

Comment:
@justindbaur did we want to make this not mutually exclusive anymore to account for Admins?